src/tools/interop/idt/capture/platform/android/android.py

```diff
@@ -0,0 +1,220 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import ipaddress
+import os
+import time
+import typing
+
+from capture.base import PlatformLogStreamer
+from capture.file_utils import create_standard_log_name
+from capture.shell_utils import BashRunner
+
+
+class Android(PlatformLogStreamer):
```

Comment:
Why aren't you depending on the adb python scripts?

https://android.googlesource.com/platform/development/+/refs/heads/main/python-packages/adb/adb/__init__.py

Reply:
No particular reason, happy to change android to this or mobly eventually if it makes sense, but I hope we can keep the general user flow the same.